pkg/smi/traffic_split.go

```diff
@@ -0,0 +1,74 @@
+package smi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	smiSplitClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+	"github.com/openservicemesh/osm-health/pkg/kuberneteshelper"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*TrafficSplitCheck)(nil)
+
+// TrafficSplitCheck implements common.Runnable
+type TrafficSplitCheck struct {
+	client      kubernetes.Interface
+	pod         *v1.Pod
+	splitClient smiSplitClient.Interface
+}
+
+// IsInTrafficSplit checks whether a pod is affected by an SMI traffic split
+func IsInTrafficSplit(client kubernetes.Interface, pod *v1.Pod, smiSplitClient smiSplitClient.Interface) TrafficSplitCheck {
+	return TrafficSplitCheck{
+		client:      client,
+		pod:         pod,
+		splitClient: smiSplitClient,
+	}
+}
+
+// Info implements common.Runnable
+func (check TrafficSplitCheck) Info() string {
+	return fmt.Sprintf("Checking whether pod %s participates in a traffic split", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check TrafficSplitCheck) Run() error {
+	services, err := kuberneteshelper.GetMatchingServices(check.client, check.pod.ObjectMeta.GetLabels(), check.pod.Namespace)
+	if err != nil {
+		log.Err(err).Msg("Error getting list of services")
+		return errors.New("error getting services")
+	}
+	trafficSplits, err := check.splitClient.SplitV1alpha2().TrafficSplits(check.pod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Err(err).Msg("Error getting list of traffic splits")
+		return errors.New("error getting traffic splits")
+	}
+	for _, trafficSplit := range trafficSplits.Items {
+		spec := trafficSplit.Spec
+		for _, backend := range spec.Backends {
+			for _, svc := range services {
+				if backend.Service == svc.Name {
+					return nil
```

Comment:
Shouldn't we group a list of traffic splits that the pod is participating in and information about that here and then return that information to the user? I'm under the impression that the user would want to know more information about how traffic flows from one pod to another and the traffic splits and services the dest pod is participating in would be helpful information in general.

Reply:
Hm that is true but for now our checks only return a pass/fail outcome (ex below). We do intend to add suggestions for failures in the future, so maybe we can also add some messaging for the passed checks. Let me open an issue to track this

![image](https://user-images.githubusercontent.com/42152676/129414493-cb249d1c-aa55-4c88-adba-36a2a6da4c53.png)

Reply:
Opened an issue to track this here: #63